prog/github/github_repository_nexus.rb

```diff
@@ -136,6 +132,14 @@ def before_run
   end
 
   label def wait
+    unless github_repository.installation.project.active?
```

Comment:
With the `nap 0`, I think we could and probably would run this block more than once. It could result in multiple semaphores accruing on the runners and repository. Potentially many semaphores if the respirate worker handling the repository is busy.

I think the safest way to handle things is to set a frame variable inside this block. If we reenter the block, `nap 5` or something similar instead of adding more semaphores.

Reply:
I think you are correct, but is there a reason to use `incr_destroy` and `nap 0` instead of using `hop_destroy`? `hop_destroy` makes it more obvious, IMO. I guess that doesn't register the deadline, though. If the deadline is the reason you are using `incr_destroy` and `nap 0`, I think it would be more understandable and more efficient to use `register_deadline` and `hop_destroy`.

Reply:
OK. If that is the current practice, this is fine.

FWIW, I think it would be better to add a `start_destroy` method, have `before_run` call that, and allow calling `start_destroy` directly from other labels. That keeps the logic in one place, and makes the calling code more understandable.